lib/countly.js

```diff
@@ -272,1928 +220,2506 @@
-            if (last - lastBeat > sessionUpdate) {
-                Countly.session_duration(last - lastBeat);
-                lastBeat = last;
+            /**
+             *  Creates new cookie or removes cookie with negative expiration
+             *  @param {String} cookieKey - The key or identifier for the store
+             *  @param {String} cookieVal - Contents of the store
+             *  @param {Number} exp - Expiration in days
+             */
+            function createCookie(cookieKey, cookieVal, exp) {
+                var date = new Date();
+                date.setTime(date.getTime() + (exp * 24 * 60 * 60 * 1000));
+                var expires = "; expires=" + date.toGMTString();
+                document.cookie = cookieKey + "=" + cookieVal + expires + "; path=/";
             }
-        }
 
-        //process event queue
-        if (eventQueue.length > 0) {
-            if (eventQueue.length <= maxEventBatch) {
-                toRequestQueue({ events: JSON.stringify(eventQueue) });
-                eventQueue = [];
-            }
-            else {
-                var events = eventQueue.splice(0, maxEventBatch);
-                toRequestQueue({ events: JSON.stringify(events) });
+            /**
+            * Returns contents of cookie
+            * @param {String} cookieKey - The key or identifier for the store
+            * @returns {Varies} stored value
+            */
+            function readCookie(cookieKey) {
+                var nameEQ = cookieKey + "=";
+                var ca = document.cookie.split(";");
+                for (var i = 0, max = ca.length; i < max; i++) {
+                    var c = ca[i];
+                    while (c.charAt(0) === " ") {
+                        c = c.substring(1, c.length);
+                    }
+                    if (c.indexOf(nameEQ) === 0) {
+                        return c.substring(nameEQ.length, c.length);
+                    }
+                }
+                return null;
             }
-            store("cly_event", eventQueue);
         }
 
-        //process request queue with event queue
-        if (!offlineMode && requestQueue.length > 0 && readyToProcess && getTimestamp() > failTimeout) {
-            readyToProcess = false;
-            var params = requestQueue[0];
-            //check if any consent to sync
-            if (hasAnyProperties(syncConsents)) {
-                if (consentTimer) {
-                    clearTimeout(consentTimer);
-                    consentTimer = null;
+        /**
+         *  Migrate from old storage to new app_key prefixed storage
+         */
+        function migrate() {
+            if (store(self.namespace + "cly_id", undefined, false, true)) {
+                //old data exists, we should migrate it
+                store("cly_id", store(self.namespace + "cly_id", undefined, false, true));
+                store("cly_event", store(self.namespace + "cly_event", undefined, false, true));
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6356797)